### DISCUSSION TITLE
Multicam setup and control over WiFi

### DISCUSSION THREAD

**Discussion by jonericcook** [gopro/OpenGoPro#366] [Q&A]
Hi 👋 

I wanted to ask a question as it pertains to my multicam use case because this conversation https://github.com/gopro/OpenGoPro/discussions/71 and the two items below have me a bit confused:

1. `The socket address for WiFi connections is 10.5.5.9:8080` - https://gopro.github.io/OpenGoPro/http_2_0#wifi-2
2. Question: `Is there currently a way to connect multiple cameras on the same Wifi network?` -- Answer: `No. Cameras can only be connected through Wi-Fi by becoming an access point itself (generating its own Wi-Fi network), not as a peripheral.` - https://gopro.github.io/OpenGoPro/faq#multi-camera-setups

For point number 1 does that mean every gopro that connects to wifi has the same IP address?

For point number 2 does that mean I can't have multiple gopros join the same wifi network?

-------

My use case / setup:
- computer
- 5Ghz wifi router that will only have my pc (connected to the back of the router via ethernet) and 4 gopros connected to it
- write a python script that will:
  -  call out to each camera (over wifi) to start recording a video
  - after N numbers of seconds call out to each camera (over wifi) to stop recording
  - call out to each camera (over wifi) to download the video that was just recorded to the computer

As a side note using USB-C doesn't appear to be an option as the gopros will be 20 ft away from the computer.

------

Thank you guys for making OpenGoPro - it's such an awesome tool 💯

**Comment by jonericcook** [comment]
@tcamise-gpsw i wanted to bump this question 🙏

**Comment by Scopi1** [comment]
Working on a similar project so if you discover any information please let me know! Right now I just iterate through the 3 cams, Capture a timestamp when all have started recording, then timestamp when recording stops. There is latency between issuing start/stops but if we have timestamp of 'real' start and 'end.' then post video processing could handle.

Reading through the API reference I found the function listed below which makes me think you can issue a command to the gopro instructing it to join a network or could this just be a typo? I'm going to dig deeper.

https://gopro.github.io/OpenGoPro/python_sdk/api.html#open_gopro.gopro_wireless.WirelessGoPro.connect_to_access_point

PS - looks like this isn't available in 0.13 but is now with 0.14.

The trouble remains for BLE. I'm really just diving into learning as I go but from what I can gather you cannot be connected to multiple BLE devices simultaniously. I did find some examples in bleak of concurrent connections to BLE devices but not sure how that could be integrated with OpenGoPro.

I apologize if none of this makes sense. I'm not a programer.

**Comment by jonericcook** [comment]
hey @Scopi1 i would check out the canon camera api https://developercommunity.usa.canon.com/s/article/Introduction-to-Camera-Control-API-CCAPI

you can have the camera join your wifi network and then control it over the api

**Reply by deenisbb** [reply]
Hi @jonericcook , 
Did you manage to get the GoPro to connect to your WiFi network to control it?

**Reply by jonericcook** [reply]
no - i went forward with a different camera company - i didnt want to buy gopros and then have things not work out

**Comment by NezRez** [comment]
I use multiple cams and going to be expanding into security and love how e z friendly gopro is so far. need a hub location and sign in spot for users  with different restrictions on their login. lets make some security/wild fire stuff

**Reply by deenisbb** [reply]
Hello,
Do you use multiple integrated cams with WiFi?

**Comment by NezRez** [comment]
using 2, connecting through tablets wifi on LE netwrok
________________________________
